src/wp-includes/post.php

```diff
@@ -3751,6 +3751,8 @@ function wp_post_mime_type_where( $post_mime_types, $table_alias = '' ) {
 function wp_delete_post( $post_id = 0, $force_delete = false ) {
 	global $wpdb;
 
+	$post_id = (int) $post_id;
```

Comment:
Since there should never be a `wp_posts` row where `ID` is `0`, then this should short-circuit.

```suggestion
	$post_id = (int) $post_id;
	if ( 0 === $post_id ) {
		return false;
	}
```